tests/test_cli.py

```diff
@@ -198,46 +201,46 @@ def test_generated(self, case, lang):
-                        # KtLint does not support absolute path in globs
-                        case_output = case_output.relative_to(Path.cwd())
-                    linter = _create_cmd(settings.linter, case_output)
-                    if ext == ".cpp":
-                        linter.append("-Wno-unused-variable")
-                        if case == "coverage":
-                            linter.append(
-                                "-Wno-null-arithmetic"
-                                if CXX == "clang++"
-                                else "-Wno-pointer-arith"
-                            )
-                    proc = run(linter, env=env)
-                    # golint is failing regularly due to exports without docs
-                    if proc.returncode and linter[0] == "golint":
-                        expect_failure = True
-                    if proc.returncode and expect_failure:
-                        raise unittest.SkipTest(f"{case}{ext} failed linter")
-                    self.assertFalse(proc.returncode)
-
-                    if expect_failure:
-                        raise AssertionError(f"{case}{ext} passed unexpectedly")
+                raise RuntimeError(f"Compiled output {exe} not detected")
+
+            self.assertTrue(stdout, "Invoked code produced no stdout")
+            stdout = stdout.splitlines()
+            self.assertEqual(expected_output, stdout)
+
+            if settings.linter and self.LINT:
+                if not spawn.find_executable(settings.linter[0]):
+                    raise unittest.SkipTest(f"{settings.linter[0]} not available")
+                if settings.ext == ".kt" and case_output.is_absolute():
+                    # KtLint does not support absolute path in globs
+                    case_output = case_output.relative_to(Path.cwd())
+                linter = _create_cmd(settings.linter, case_output)
+                if ext == ".cpp":
+                    linter.append("-Wno-unused-variable")
+                    if case == "coverage":
+                        linter.append(
+                            "-Wno-null-arithmetic"
+                            if CXX == "clang++"
+                            else "-Wno-pointer-arith"
+                        )
+                proc = run(linter, env=env)
+                # golint is failing regularly due to exports without docs
+                if proc.returncode and linter[0] == "golint":
+                    expect_failure = True
+                if proc.returncode and expect_failure:
+                    raise unittest.SkipTest(f"{case}{ext} failed linter")
+                self.assertFalse(proc.returncode)
+
+                if expect_failure:
+                    raise AssertionError(f"{case}{ext} passed unexpectedly")
 
         finally:
             if not self.KEEP_GENERATED:
                 case_output.unlink(missing_ok=True)
-            exe.unlink(missing_ok=True)
+                exe.unlink(missing_ok=True)
```

Comment:
So KEEP_GENERATED means keep the generated source as well as the binary.
Sounds good. Wanted to make sure this was intentional.

Reply:
Ya, occasionally I need to run the exe afterwards to see why it is failing.